R/insert-ui.R

```diff
@@ -172,3 +172,275 @@ removeUI <- function(selector,
   if (!immediate) session$onFlushed(callback, once = TRUE)
   else callback()
 }
+
+
+#' Dynamically add a tabPanel
+#'
+#' Dynamically add a \code{tabPanel()} into an existing \code{tabsetPanel}.
+#'
+#' @param tabsetPanelId The \code{id} of the \code{tabsetPanel()} into which
+#'   \code{tab} will be inserted.
+#'
+#' @param tab The tab element to be added (must be created with
+#'   \code{tabPanel}).
+#'
+#' @param target The \code{value} of an existing \code{tabPanel()}, next to
+#'   which \code{tab} will be added. If \code{NULL},the \code{tab} will be
+#'   placed either as the first tab or the last tab, depending on the
+#'   \code{position} argument.
```

Comment:
So, in your proposed way, what does `left` mean when target is `NULL`? Is it "_to the left of nothing_", meaning the rightmost tab? Or is it "_to the left of nothing_", meaning the leftmost tab just the same?

Regardless, I also wanted to point out something. I chose to use this combination of `target` and `position` keeping in mind that the most important things would be:

1. All possible positions must be covered
2. The most typical use cases should be the most intuitive. There isn't much challenging about `position` is `target` is non-NULL. However, if `target` is NULL (or whatever value we decide to give to mean that the target is the whole tabset), then we want to allow users to quickly specify whether they want to put the tab at the beginning or at the end (which I think are the only two reasonable options when dealing with the whole tabset as the "target"). So, in this way, `right` means "to the right of all other tabs" and `left` means to the "left of all other tabs". This is particularly important when you want to keep either prepending or appending tabs, but you don't know (or don't want to keep track) of the tab value of the first or last tab, respectively.

So, my question is, under your scheme, is there a way that the user can prepend/append tabs without referring to specific tab names? That was the biggest consideration for me. So, while I agree that it is not ideal to have two ways of doing the same thing, I think I care more about the symmetry in being able to both prepend or append tabs without having to know what's the first (or last) tab value?

Let me know what you think :)